exp/api/v1beta1/gcpmanagedmachinepool_types.go

```diff
@@ -37,6 +37,35 @@ type GCPManagedMachinePoolSpec struct {
 	// Scaling specifies scaling for the node pool
 	// +optional
 	Scaling *NodePoolAutoScaling `json:"scaling,omitempty"`
+	// NodeLocations is the list of zones in which the NodePool's
+	// nodes should be located.
+	// +optional
+	NodeLocations []string `json:"nodeLocations,omitempty"`
+	// ImageType is image type to use for this nodepool.
+	// +optional
+	ImageType *string `json:"imageType,omitempty"`
+	// InstanceType is name of Compute Engine machine type.
+	// +optional
+	InstanceType *string `json:"instanceType,omitempty"`
+	// DiskType is type of the disk attached to each node.
+	// +kubebuilder:validation:Enum=pd-standard;pd-ssd;pd-balanced
+	// +optional
+	DiskType *string `json:"diskType,omitempty"`
+	// DiskSizeGB is size of the disk attached to each node,
+	// specified in GB.
+	// +optional
+	DiskSizeGB *int64 `json:"diskSizeGB,omitempty"`
```

Comment:
Should we have a min value validation on this as well to stop people entering a negative or zero value?